src/safeds/data/image/containers/_image.py

```diff
@@ -139,13 +207,13 @@ def to_png_file(self, path: str | Path) -> None:
             The path to the PNG file.
         """
         Path(path).parent.mkdir(parents=True, exist_ok=True)
-        self._image.save(path, format="png")
+        save_image(self._image_tensor.to(torch.float32) / 255, path, format="png")
 
     # ------------------------------------------------------------------------------------------------------------------
     # IPython integration
     # ------------------------------------------------------------------------------------------------------------------
```

Comment:
Something is not right here: `_repr_jpeg_` and `_repr_png_` were used for the IPython integration. `__eq__` is just the normal operator overload (was wrong before).